### PR TITLE
Remove use of install_katello_ca with register_contenthost

### PR DESCRIPTION
**Pull request by Griffin-Sullivan**
### Problem Statement
We will be removing the katello-ca-consumer RPM by default. Our automation should only use global registration anyway.

### Solution
Swap uses of `install_katello_ca` and `register_contenthost` to the `register` method which uses global registration. Use default location and a basic activation key where needed.

### PRT test Cases example
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation

### CODE REVIEW

**Comment by Griffin-Sullivan** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7368
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation --external-logging
Test Result : ========== 11 failed, 3631 warnings, 14 errors in 18089.14s (5:01:29) ==========
```

**Comment by Griffin-Sullivan** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7390
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation --external-logging
Test Result : ===== 10 failed, 6 passed, 4010 warnings, 9 errors in 20144.39s (5:35:44) ======
```

**Comment by Griffin-Sullivan** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7596
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation --external-logging
Test Result : ===== 12 failed, 10 passed, 4224 warnings, 3 errors in 21100.35s (5:51:40) =====
```

**Comment by Griffin-Sullivan** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7731
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation --external-logging
Test Result : ===== 16 failed, 5 passed, 4219 warnings, 7 errors in 21217.24s (5:53:37) ======
```

**Comment by Griffin-Sullivan** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7752
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation --external-logging
Test Result : ===== 12 failed, 9 passed, 3896 warnings, 7 errors in 19272.31s (5:21:12) ======
```

**Comment by Griffin-Sullivan** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7758
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation --external-logging
Test Result : ===== 8 failed, 13 passed, 3795 warnings, 7 errors in 19332.91s (5:22:12) ======
```

**Comment by Griffin-Sullivan** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7761
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy --external-logging
Test Result : ============ 2 failed, 4 passed, 468 warnings in 3015.54s (0:50:15) ============
```

**Comment by Griffin-Sullivan** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7765
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation --external-logging
Test Result : ================= 30 warnings, 25 errors in 295.41s (0:04:55) ==================
```

**Comment by Griffin-Sullivan** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7768
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation --external-logging
Test Result : ===== 6 failed, 11 passed, 2647 warnings, 11 errors in 14552.13s (4:02:32) =====
```

**Comment by Griffin-Sullivan** [comment]
Ok looking through the failures it seems these are not related to the changes and have been failing for the same reasons according to Ibutsu.

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation

File: robottelo/hosts.py [hunk 1]
```diff
@@ -1155,17 +1098,10 @@ def configure_rhai_client(
...
+        # Ensure insights-client rpm is installed
         if self.execute('yum install -y insights-client').status != 0:
             raise ContentHostError('Unable to install insights-client rpm')
 
-        if register_insights:
-            # Register client
-            if self.execute('insights-client --register').status != 0:
-                raise ContentHostError('Unable to register client to Insights through Satellite')
-            if self.execute('insights-client --test-connection').status != 0:
-                raise ContentHostError('Test connection failed via insights.')
```
Comment (vsedmik) [inline robottelo/hosts.py:1167]: Why are we removing this part? Looks like we are using it, `register_insights` is set true by default.
WDYT @ColeHiggins2 ?

Comment (Griffin-Sullivan) [reply robottelo/hosts.py:1167]: The `setup_insights` param in `register` on L1079 will install and register insights-client if the user passes `register_insights=True`

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7911
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_host.py::test_positive_install_package_via_rex tests/foreman/cli/test_http_proxy.py::test_insights_client_registration_with_http_proxy tests/foreman/ui/test_subscription.py::test_positive_view_vdc_subscription_products tests/foreman/destructive/test_host.py::TestHostCockpit::test_positive_cockpit tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_custom_loc tests/foreman/cli/test_contentview.py::TestContentView::test_positive_sub_host_with_restricted_user_perm_at_default_loc tests/foreman/destructive/test_capsule_loadbalancer.py::test_loadbalancer_install_package tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end tests/foreman/endtoend/test_cli_endtoend.py::test_positive_cli_end_to_end tests/foreman/installer/test_installer.py::test_capsule_installation --external-logging
Test Result : ===== 4 failed, 18 passed, 3937 warnings, 3 errors in 20392.35s (5:39:52) ======
```

**Review by pondrejk — APPROVED** [review]
ack for the Endeavour's components

**Review by jameerpathan111 — APPROVED** [review]

File: robottelo/hosts.py (hunk 1 above)
Comment (vsedmik) [reply robottelo/hosts.py:1167]: Oh I overlooked that, then we are good I guess. :+1:

**Review by ogajduse — COMMENTED** [review]
Next steps (future PR) I see are:
- remove `install_katello_ca` method
- add `if reg_result.status == 0: self._satellite = target` at the very end of the `register` and `api_register` methods

**Pull request merged**